coalib/results/Diff.py

```diff
@@ -70,6 +72,90 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        :raises RuntimeError: Raised when the context lines or the
+                              lines to be removed do not match in
+                              the original file and the unified diff.
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        if not patch_set:
+            return diff
+
+        file = patch_set[0]
+
+        for hunk in file:
+            file_line = hunk.source_start
+            hunk_iterator = iter(hunk)
+
+            try:
+                while True:
+                    line = next(hunk_iterator)
+                    source_code = str(line)[1:]
+                    if line.is_added:
+                        add_set = []
+                        # As ``Diff`` does not allow line additions to a
+                        # position more than one time, add all the
+                        # consecutive '+' lines at once.
+
+                        try:
+                            while line.is_added:
+                                add_set.append(source_code)
+                                line = next(hunk_iterator)
+                                source_code = str(line)[1:]
+
+                            diff.add_lines(file_line-1, add_set)
+
+                        except StopIteration:
+                            diff.add_lines(file_line-1, add_set)
+                            break
+
+                    if line.is_removed:
+                        if (source_code !=
+                                original_file[file_line-1].strip('\n')):
+                            raise RuntimeError(
+                                'The line to delete does not match with '
+                                'the line in the original file. '
+                                'Line to delete: {!r}, '
+                                'Original line #{!r}: {!r}'.format(
+                                    source_code,
+                                    file_line,
+                                    original_file[file_line-1].strip('\n'))
+                                )
+                        diff.delete_line(file_line)
+                        file_line += 1
+
+                    else:
+                        if (source_code !=
+                                original_file[file_line-1].strip('\n')):
+                            raise RuntimeError(
+                                'Context lines do not match. '
+                                'Line from unified diff: {!r}, '
+                                'Original line #{!r}: {!r}'.format(
+                                    source_code,
+                                    file_line,
+                                    original_file[file_line-1].strip('\n'))
+                                )
+                        file_line += 1
```

Comment:
we should move this out of the if, as for both branches you execute `file_line += 1`